app/controllers/MovementsController.scala

```diff
@@ -20,87 +20,166 @@ import java.time.OffsetDateTime
 
 import connectors.MessageConnector
 import controllers.actions.AuthenticateActionProvider
+import controllers.actions.AuthenticatedGetArrivalForWriteActionProvider
+import controllers.actions.AuthenticatedGetOptionalArrivalForWriteActionProvider
+import controllers.actions.LockActionProvider
 import javax.inject.Inject
+import models.MessageReceived.ArrivalSubmitted
+import models.Arrival
+import models.ArrivalId
 import models.Arrivals
+import models.MessageReceived
+import models.MessageSender
+import models.MessageState
 import models.MessageType
 import models.SubmissionResult
 import models.TransitWrapper
+import models.request.ArrivalRequest
 import play.api.Logger
 import play.api.libs.json.Json
 import play.api.mvc.Action
 import play.api.mvc.AnyContent
 import play.api.mvc.ControllerComponents
+import play.api.mvc.DefaultActionBuilder
 import repositories.ArrivalMovementRepository
 import services.ArrivalMovementService
-import uk.gov.hmrc.http.BadGatewayException
-import uk.gov.hmrc.http.GatewayTimeoutException
-import uk.gov.hmrc.http.ServiceUnavailableException
+import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.play.bootstrap.controller.BackendController
 
 import scala.concurrent.ExecutionContext
 import scala.concurrent.Future
+import scala.util.Failure
+import scala.util.Success
 import scala.xml.NodeSeq
 
 class MovementsController @Inject()(
+  defaultActionBuilder: DefaultActionBuilder,
   cc: ControllerComponents,
   arrivalMovementRepository: ArrivalMovementRepository,
   arrivalMovementService: ArrivalMovementService,
+  authenticatedOptionalArrival: AuthenticatedGetOptionalArrivalForWriteActionProvider,
   authenticate: AuthenticateActionProvider,
+  authenticateForWrite: AuthenticatedGetArrivalForWriteActionProvider,
   messageConnector: MessageConnector
 )(implicit ec: ExecutionContext)
     extends BackendController(cc) {
 
   private val logger = Logger(getClass)
 
-  def createMovement: Action[NodeSeq] = authenticate().async(parse.xml) {
+  def createMovement: Action[NodeSeq] = authenticatedOptionalArrival().async(parse.xml) {
```

Comment:
Noticed we have multiple authentication actions. Does this indicate that we may need multiple controllers rather than just a single one that handles every action

Reply:
I'll look at splitting them out